genshin/client/components/hoyolab.py

```diff
@@ -29,12 +29,19 @@ async def search_users(
         )
         return [models.PartialHoyolabUser(**i["user"]) for i in data["list"]]
 
-    async def get_hoyolab_user(self, hoyolab_id: int, *, lang: typing.Optional[str] = None) -> models.FullHoyolabUser:
+    async def get_hoyolab_user(
+        self, hoyolab_id: typing.Optional[int] = None, *, lang: typing.Optional[str] = None
+    ) -> models.FullHoyolabUser:
         """Get a hoyolab user."""
-        data = await self.request_hoyolab(
-            "https://bbs-api-os.hoyolab.com/community/painter/wapi/user/full",
+        if hoyolab_id:
+            params = dict(uid=hoyolab_id)
+        else:
+            params = None
+
+        data = await self.request_bbs(
+            url=routes.BBS_GET_USER_URL.get_url(self.region),
             lang=lang,
-            params=dict(uid=hoyolab_id),
+            params=params,
```

Comment:
wouldn't just `params=dict(uid=hoyolab_id or "")` work? I know many endpoints treat an empty field the same as if the field was not given.

if not use `params=dict(uid=hoyolab_id) if hoyolab_id else None`

Reply:
Fixed.